### PR TITLE
[zh-CN] sync  `Getting started` part of  `web-ext`

### PR DESCRIPTION
**Pull request by uiolee**
### Description

- Synchronize with content
    - Mainly to add missing text.
    - keep their structure the same.
    - correct outdated links.

### Motivation

Some articles are missing something compared to CONTENT. I don't think readers want to view both versions of the article and compare them carefully. I think an incomplete document is incompetent. So I added these contents from CONTENT to our Chinese documentation, although they have not been translated yet.

### CODE REVIEW

File: files/zh-cn/mozilla/add-ons/webextensions/anatomy_of_a_webextension/index.md [hunk 1]
```diff
@@ -19,9 +19,9 @@ slug: Mozilla/Add-ons/WebExtensions/Anatomy_of_a_WebExtension
 - [内容脚本](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Anatomy_of_a_WebExtension#内容脚本): 扩展中包含的 JavaScript，你可以注入它们到网页中。
 - [网络可访问资源](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Anatomy_of_a_WebExtension#网络可访问资源): 使打包好的内容可用于网页与内容脚本中。
 
-![](webextension-anatomy.png)
+![The components of a web extension. The manifest.JSON must be present in all extensions. It provides pointers to background pages, content scripts, browser actions, page actions, options pages, and web accessible resources. Background pages consist of HTML and JS. Content scripts consist of JS and CSS. The user clicks on an icon to trigger browser actions and page actions and the resulting pop-up consists of HTML, CSS, and JS. Options pages consist of HTML, CSS, and JS.](webextension-anatomy.png)
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/anatomy_of_a_webextension/index.md:22]: please translate the sentences

File: files/zh-cn/mozilla/add-ons/webextensions/examples/index.md [hunk 2]
```diff
@@ -9,11 +9,12 @@ slug: Mozilla/Add-ons/WebExtensions/Examples
 
 这些示例工作在 Firefox Nightly 下：大部分示例在 Firefox 的早期版本中也能够运行，请检查扩展 manifest.json 文件中的 [strict_min_version](/zh-CN/Add-ons/WebExtensions/manifest.json/applications) 键。
 
+> **Warning:** Some examples work only on specific domains or pages. Details of any restrictions are provided in each example's readme file. None of the examples work in private browsing windows by default, see [Extensions in Private Browsing](https://support.mozilla.org/zh-CN/kb/extensions-private-browsing#w_enabling-or-disabling-extensions-in-private-windows) for details.
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/examples/index.md:12]: same

File: files/zh-cn/mozilla/add-ons/webextensions/what_are_webextensions/index.md [hunk 3]
```diff
@@ -11,25 +11,25 @@ slug: Mozilla/Add-ons/WebExtensions/What_are_WebExtensions
 
 **提升或者补充网站功能**: 利用扩展来实现额外的浏览器内特性或者来自你网站的信息。允许用户搜集他们访问过的页面细节来提升你所提供的服务。
 
-![](amazon_add_on.png)
+![Amazon add-on example providing price comparison features](amazon_add_on.png)
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/what_are_webextensions/index.md:14]: same

**Review by yin1999 — COMMENTED** [review]
Hi @uiolee, please translate those English sentences. **Do not** add more English words in zh-CN translation. Thanks

File: files/zh-cn/mozilla/add-ons/webextensions/anatomy_of_a_webextension/index.md (hunk 1 above)
Comment (BSPR0002) [inline files/zh-cn/mozilla/add-ons/webextensions/anatomy_of_a_webextension/index.md:22]:
```suggestion
![Web 扩展的组件。manifest.JSON 是所有扩展里都必须存在的。它提供指向后台页面、内容脚本、浏览器操作、页面操作、选项页和网络可访问资源的指针。后台页面由 HTML 和 JS 组成。内容脚本由 JS 和 CSS 组成。用户单击扩展图标能够触发浏览器操作和页面操作，由此产生的弹出窗口由 HTML、CSS 和 JS 组成。选项页由 HTML、CSS 和 JS 组成。](webextension-anatomy.png)
```
比起这个，不更应该把这张图片给改改么

File: files/zh-cn/mozilla/add-ons/webextensions/what_next_/index.md [hunk 4]
```diff
@@ -47,5 +49,16 @@ slug: Mozilla/Add-ons/WebExtensions/What_next_
 
 - 更多关于浏览器扩展背后的基本概念，从如何[使用 JavaScript API](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/API) 的细节开始。
 - 可供浏览器扩展使用的[用户界面组件](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/user_interface)的指南。
+- A collection of how-to guides on achieving key tasks in your extensions or making use of the JavaScript APIs.
```
Comment (BSPR0002) [inline files/zh-cn/mozilla/add-ons/webextensions/what_next_/index.md:52]:
```suggestion
- 一系列关于如何在扩展中实现关键任务与运用 JavaScript API 的指南。
```

File: files/zh-cn/mozilla/add-ons/webextensions/what_next_/index.md [hunk 5]
```diff
@@ -47,5 +49,16 @@ slug: Mozilla/Add-ons/WebExtensions/What_next_
...
+
+You'll also want to head on over to Extension Workshop where you'll find everything you need to know about creating extensions for Firefox, including:
+
+- [an overview of the Firefox extension features](https://extensionworkshop.com/#about)
+- [details of the tools and processes for developing and testing](https://extensionworkshop.com/documentation/develop/)
+- [how to publish your extension on addons.mozilla.org or distribute it yourself](https://extensionworkshop.com/documentation/publish/)
+- [how to manage your published extension](https://extensionworkshop.com/documentation/manage/)
+- [an enterprise guide for developing and using extensions](https://extensionworkshop.com/documentation/enterprise/)
+- [how to develop themes for Firefox](https://extensionworkshop.com/documentation/themes/)
+- [details about the Firefox developer communities](https://extensionworkshop.com/community/)
```
Comment (BSPR0002) [inline files/zh-cn/mozilla/add-ons/webextensions/what_next_/index.md:64]:
```suggestion
你还能在扩展工坊找到创建 Firefox 扩展所需要了解的所有东西，包括：

- [Firefox 扩展特性一览](https://extensionworkshop.com/#about)
- [用于开发及测试的工具和工序的详细介绍](https://extensionworkshop.com/documentation/develop/)
- [如何在 addons.mozilla.org 上发布你的扩展以及如何自行分发扩展](https://extensionworkshop.com/documentation/publish/)
- [如何管理你发布的扩展](https://extensionworkshop.com/documentation/manage/)
- [面向企业的开发与使用扩展的指南](https://extensionworkshop.com/documentation/enterprise/)
- [如何为 Firefox 开发主题](https://extensionworkshop.com/documentation/themes/)
- [Firefox 开发者社区的详细介绍](https://extensionworkshop.com/community/)
```

File: files/zh-cn/mozilla/add-ons/webextensions/your_second_webextension/index.md [hunk 6]
```diff
@@ -402,7 +401,11 @@ content script 做的第一件事是检查全局变量 `window.hasRun`：如果
 
 创建"beasts"文件夹，之后将图片放入并命名。你可以从 [GitHub 仓库](https://github.com/mdn/webextensions-examples/tree/main/beastify/beasts)或这里下载图片：
 
-![](frog.jpg)![](snake.jpg)![](turtle.jpg)
+![A brown frog.](frog.jpg)
+
+![An emerald tree boa with white stripes.](snake.jpg)
+
+![A red-eared slider turtle.](turtle.jpg)
```
Comment (BSPR0002) [inline files/zh-cn/mozilla/add-ons/webextensions/your_second_webextension/index.md:408]:
```suggestion
![一只棕色青蛙。](frog.jpg)

![一条有白色条纹的翡翠树蚺。](snake.jpg)

![一只巴西龟。](turtle.jpg)
```

File: files/zh-cn/mozilla/add-ons/webextensions/your_second_webextension/index.md [hunk 7]
```diff
@@ -437,11 +440,11 @@ Firefox 45 开始，你可以临时从硬盘中安装扩展
 
 然后你应该已经看到扩展图标出现在了 Firefox 的工具条上：
 
-{{EmbedYouTube("sAM78GU4P34")}}
+![The beastify icon in the Firefox toolbar](beastify_icon.png)
```
Comment (BSPR0002) [inline files/zh-cn/mozilla/add-ons/webextensions/your_second_webextension/index.md:443]:
```suggestion
![Firefox 工具栏上的 beastify 扩展图标](beastify_icon.png)
```

File: files/zh-cn/mozilla/add-ons/webextensions/your_second_webextension/index.md [hunk 8]
```diff
@@ -437,11 +440,11 @@ Firefox 45 开始，你可以临时从硬盘中安装扩展
...
 
 打开一个网页，然后点击图标，选择一个动物，然后观察网页的变化
 
-{{EmbedYouTube("YMQXyAQSiE8")}}
+![A page replaced with the image of a turtle](beastify_page.png)
```
Comment (BSPR0002) [inline files/zh-cn/mozilla/add-ons/webextensions/your_second_webextension/index.md:447]:
```suggestion
![被替换成乌龟图像的页面。](beastify_page.png)
```

File: files/zh-cn/mozilla/add-ons/webextensions/your_second_webextension/index.md [hunk 9]
```diff
@@ -451,3 +454,12 @@ Firefox 45 开始，你可以临时从硬盘中安装扩展
...
 ```
+
+## What's next?
+
+Now that you've created a more advanced WebExtension for Firefox:
+
+- [read about the anatomy of an extension](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Anatomy_of_a_WebExtension)
+- [explore the extension examples](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Examples)
+- [find out what you need to develop, test, and publish your extension](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/What_next_)
+- [take your learning further](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/What_next_#continue_your_learning_experience).
```
Comment (BSPR0002) [inline files/zh-cn/mozilla/add-ons/webextensions/your_second_webextension/index.md:465]:
```suggestion
## 接下来做什么？

你已经创建了一个更加高级的 Firefox Web 扩展，接下来可以：

- [阅读扩展剖析](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Anatomy_of_a_WebExtension)
- [浏览扩展示例](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Examples)
- [学习如何开发、测试、发布你的扩展](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/What_next_)
- [进一步学习](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/What_next_#continue_your_learning_experience).
```

**Review by BSPR0002 — CHANGES_REQUESTED** [review]

**Comment by uiolee** [comment]
Hi @yin1999 . My thought is to pull content first, and then our readers and volunteers will see it. Then that content will be translated. Otherwise, we may ignore these content. I don't know if it fits the workflow.

**Comment by BSPR0002** [comment]
> Hi @yin1999 . My thought is to pull content first, and then our readers and volunteers will see it. Then that content will be translated. Otherwise, we may ignore these content. I don't know if it fits the workflow.

IMO, this is most about the accessibility of the images, the absence of these contents in the original version may have been because they are not typically visible, leading to a lack of effort in translation. However, this is not considered good practice. Though you may not a chinese speaker, I think your approach is more like passing trouble to others.

**Comment by yin1999** [comment]
> Hi @yin1999 . My thought is to pull content first, and then our readers and volunteers will see it. Then that content will be translated. Otherwise, we may ignore these content. I don't know if it fits the workflow.

The problem is, we have too many document that are not fully translated, and many documents are even fully in English. So I don't like the idea of directly adding English content to l10n. If a document is out-of-date (see the "last modified" at the bottom of page), it's better to report an issue rather than mixin English content, as Chrome could not perform the translation well with mixin content (The Chinese words in document may be "mis-translated", we just receive an [issue about this](https://github.com/mdn/translated-content/issues/14020)).

**Comment by uiolee** [comment]
> 
Thank you for your answer. I think I understand now.

File: files/zh-cn/mozilla/add-ons/webextensions/examples/index.md [hunk 10]
```diff
@@ -9,11 +9,12 @@ slug: Mozilla/Add-ons/WebExtensions/Examples
...
 如果你想尝试这些示例，有下面三个选择：
 
-1. 克隆代码库，通过 ["临时载入附加组件"](/zh-CN/Add-ons/WebExtensions/Temporary_Installation_in_Firefox) 功能直接从源文件夹中载入扩展组件。再重启浏览器前该扩展将一直存在。
-2. 克隆代码库，使用 [web-ext](/zh-CN/Add-ons/WebExtensions/Getting_started_with_web-ext) 命令行工具运行安装了扩展的 Firefox。
-3. 克隆代码库，进入 [build](https://github.com/mdn/webextensions-examples/tree/main/build) 文件夹。该文件夹包含所有示例的已构建、带签名版本。你可以在 Firefox 中打开（菜单栏->文件->打开文件）并且永久安装这些扩展。这些扩展和直接从 [addons.mozilla.org](https://addons.mozilla.org/en-US/firefox/) 安装的扩展别无二致。
+1. 克隆代码库，通过 ["临时载入附加组件"](https://extensionworkshop.com/documentation/develop/temporary-installation-in-firefox/) 功能直接从源文件夹中载入扩展组件。再重启浏览器前该扩展将一直存在。
+2. 克隆代码库，使用 [`web-ext`](https://extensionworkshop.com/documentation/develop/getting-started-with-web-ext/) 命令行工具运行安装了扩展的 Firefox。
+3. 克隆代码库，进入 [`build`](https://github.com/mdn/webextensions-examples/tree/main/build) 文件夹。该文件夹包含所有示例的已构建、带签名版本。你可以在 Firefox 中打开（菜单栏->文件->打开文件）并且永久安装这些扩展。这些扩展和直接从 [addons.mozilla.org](https://addons.mozilla.org/zh-CN/firefox/) 安装的扩展别无二致。
```
[the remark this answers is not in the document]
Comment (uiolee) [reply files/zh-cn/mozilla/add-ons/webextensions/examples/index.md]: yep

File: files/zh-cn/mozilla/add-ons/webextensions/examples/index.md [hunk 11]
```diff
@@ -7,16 +7,18 @@ slug: Mozilla/Add-ons/WebExtensions/Examples
 
 为帮助了解如何开发扩展，我们维护了一个包含简单扩展示例的代码仓库在<https://github.com/mdn/webextensions-examples>. 本文描述了仓库代码中所使用的 WebExtension APIs。
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/examples/index.md:8]:
```suggestion
为帮助了解如何开发扩展，我们维护了一个包含简单扩展示例的代码仓库在<https://github.com/mdn/webextensions-examples>。本文描述了仓库代码中所使用的 WebExtension API。
```

File: files/zh-cn/mozilla/add-ons/webextensions/examples/index.md [hunk 12]
```diff
@@ -7,16 +7,18 @@ slug: Mozilla/Add-ons/WebExtensions/Examples
...
 如果你想尝试这些示例，有下面三个选择：
 
-1. 克隆代码库，通过 ["临时载入附加组件"](/zh-CN/Add-ons/WebExtensions/Temporary_Installation_in_Firefox) 功能直接从源文件夹中载入扩展组件。再重启浏览器前该扩展将一直存在。
-2. 克隆代码库，使用 [web-ext](/zh-CN/Add-ons/WebExtensions/Getting_started_with_web-ext) 命令行工具运行安装了扩展的 Firefox。
-3. 克隆代码库，进入 [build](https://github.com/mdn/webextensions-examples/tree/main/build) 文件夹。该文件夹包含所有示例的已构建、带签名版本。你可以在 Firefox 中打开（菜单栏->文件->打开文件）并且永久安装这些扩展。这些扩展和直接从 [addons.mozilla.org](https://addons.mozilla.org/en-US/firefox/) 安装的扩展别无二致。
+1. 克隆代码库，通过[“临时载入附加组件”](https://extensionworkshop.com/documentation/develop/temporary-installation-in-firefox/)功能直接从源文件夹中载入扩展组件。在重启浏览器前，该扩展将一直存在。
+2. 克隆代码库，使用 [`web-ext`](https://extensionworkshop.com/documentation/develop/getting-started-with-web-ext/) 命令行工具运行安装了扩展的 Firefox。
+3. 克隆代码库，在 Firefox 中使用**文件**>**打开文件**，在 [`build`](https://github.com/mdn/webextensions-examples/tree/main/build) 文件夹中找到这些例子。`build`文件夹包含所有例子的构建和签名版本。这样就可以永久地安装这些例子了。
 
 > **警告：** 不要提交这些关于 WebExtension 例子的示例到 AMO (addons.mozilla.org) 上去，你无需为附加组件签名就能运行 WebExtension 的示例。只要按照上面的步骤来就可以了。
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/examples/index.md:20]:
```suggestion
> **警告：** 不要提交这些关于 WebExtension 例子的示例到 AMO（addons.mozilla.org）上去，你无需为附加组件签名就能运行 WebExtension 的示例。只要按照上面的步骤来就可以了。
```

File: files/zh-cn/mozilla/add-ons/webextensions/what_are_webextensions/index.md [hunk 13]
```diff
@@ -11,25 +11,25 @@ slug: Mozilla/Add-ons/WebExtensions/What_are_WebExtensions
 
 **提升或者补充网站功能**: 利用扩展来实现额外的浏览器内特性或者来自你网站的信息。允许用户搜集他们访问过的页面细节来提升你所提供的服务。
 
-![](amazon_add_on.png)
+![图中的 Amazon 扩展提供比价功能](amazon_add_on.png)
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/what_are_webextensions/index.md:14]: screen reader would "tell" the disables that "this is an image", please provide the description only. And in Firefox, we translate "add-on" into "附加组件".

```suggestion
![Amazon 附加组件示例提供比价功能](amazon_add_on.png)
```

File: files/zh-cn/mozilla/add-ons/webextensions/what_are_webextensions/index.md [hunk 14]
```diff
@@ -11,25 +11,25 @@ slug: Mozilla/Add-ons/WebExtensions/What_are_WebExtensions
...
 
 示例：[亚马逊助手](https://addons.mozilla.org/zh-CN/firefox/addon/amazon-browser-bar/)，[OneNote Web Clipper](https://addons.mozilla.org/zh-CN/firefox/addon/onenote-clipper/)，[Grammarly for Firefox](https://addons.mozilla.org/zh-CN/firefox/addon/grammarly-1/)
 
 **让用户展现他们的个性**: 浏览器扩展可以操控网页的内容；例如，让用户在每个他们访问的页面上添加最喜欢的 logo 或者图片作为背景。扩展也可以让用户更新 Firefox 的界面，就像独立的[主题扩展](https://extensionworkshop.com/documentation/themes/)一样。
 
-![](myweb_new_tab_add_on.png)
+![图中的 My Web New Tab 扩展展现蝙蝠侠主题](myweb_new_tab_add_on.png)
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/what_are_webextensions/index.md:20]:
```suggestion
![My Web New Tab 附加组件展现蝙蝠侠主题](myweb_new_tab_add_on.png)
```

File: files/zh-cn/mozilla/add-ons/webextensions/what_are_webextensions/index.md [hunk 15]
```diff
@@ -11,25 +11,25 @@ slug: Mozilla/Add-ons/WebExtensions/What_are_WebExtensions
...
 
 示例：[MyWeb New Tab](https://addons.mozilla.org/zh-CN/firefox/addon/myweb-new-tab/), [Tabliss](https://addons.mozilla.org/zh-CN/firefox/addon/tabliss/), 和 [VivaldiFox](https://addons.mozilla.org/zh-CN/firefox/addon/vivaldifox/)
 
 **从网页中添加或者删除内容**: 你可能想要帮助用户从网页中阻止一些侵扰的广告，当网页中提到一个国家或者城市的时候提供旅游指南，或者重组页面的内容来提供一个连续的阅读体验。利用可以访问和更新一个页面里的 HTML 和 CSS 的能力，扩展可以帮助用户以他们想要的形式来查看网页。
 
-![](ublock_origin_add_on.png)
+![图中的 uBlock origin 扩展能够拦截跟踪器统计](ublock_origin_add_on.png)
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/what_are_webextensions/index.md:26]:
```suggestion
![uBlock origin 附加组件示例和拦截跟踪器统计](ublock_origin_add_on.png)
```

File: files/zh-cn/mozilla/add-ons/webextensions/what_are_webextensions/index.md [hunk 16]
```diff
@@ -11,25 +11,25 @@ slug: Mozilla/Add-ons/WebExtensions/What_are_WebExtensions
...
 
 示例：[uBlock Origin](https://addons.mozilla.org/zh-CN/firefox/addon/ublock-origin/), [Reader](https://addons.mozilla.org/zh-CN/firefox/addon/reader/), 和 [Toolbox for Google Play Store™](https://addons.mozilla.org/zh-CN/firefox/addon/toolbox-google-play-store/)
 
 **添加工具和新的浏览特性**: 给任务面板添加新特性，或者从 URL 地址，超链接，或者页面文字生成二维码。有了灵活的界面选项和 [WebExtensions APIs](/zh-CN/docs/Mozilla/Add-ons/WebExtensions) 的能力，你可以很容易的向浏览器添加新的特性。并且，你可以改善几乎任何网站的特性和功能，而不必是你自己的网站。
 
-![](qr_code_image_generator_add_on.png)
+![图中的扩展是一个二维码生成器](qr_code_image_generator_add_on.png)
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/what_are_webextensions/index.md:32]:
```suggestion
![二维码生成器附加组件示例](qr_code_image_generator_add_on.png)
```

File: files/zh-cn/mozilla/add-ons/webextensions/what_are_webextensions/index.md [hunk 17]
```diff
@@ -41,7 +41,7 @@ slug: Mozilla/Add-ons/WebExtensions/What_are_WebExtensions
 
 **添加开发工具**: 你可以为你的业务提供网站开发工具或者开发一个有用的技术或者你想分享的网站开发方法。无论如何，你可以通过在开发者工具栏上添加一个新的标签来增强内置的 Firefox 开发者工具。
 
-![](axe_developer_tools_add_on.png)
+![图中是 axe 无障碍功能测试扩展](axe_developer_tools_add_on.png)
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/what_are_webextensions/index.md:44]:
```suggestion
![axe 无障碍功能测试附加组件示例](axe_developer_tools_add_on.png)
```

File: files/zh-cn/mozilla/add-ons/webextensions/your_second_webextension/index.md [hunk 18]
```diff
@@ -451,3 +454,12 @@ Firefox 45 开始，你可以临时从硬盘中安装扩展
...
 ```
+
+## 接下来做什么？
+
+你已经创建了一个更加高级的 Firefox Web 扩展，接下来可以：
+
+- [阅读扩展剖析](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Anatomy_of_a_WebExtension)
+- [浏览扩展示例](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Examples)
+- [学习如何开发、测试、发布你的扩展](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/What_next_)
+- [进一步学习](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/What_next_#continue_your_learning_experience).
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/your_second_webextension/index.md:465]:
```suggestion
- [进一步学习](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/What_next_#继续你的学习经历)。
```

**Review by yin1999 — APPROVED** [review]
Thanks

**Pull request merged**